generator/internal/rust/codec.go

```diff
@@ -925,9 +925,33 @@ func derefFieldPath(fieldPath string, message *api.Message, state *api.APIState)
 	return expression.String()
 }
 
+func typezFieldPath(fieldPath string, message *api.Message, state *api.APIState) api.Typez {
+	typez := api.UNDEFINED_TYPE
+	components := strings.Split(fieldPath, ".")
+	msg := message
+	for _, name := range components {
+		if msg == nil {
+			slog.Error("cannot build full expression", "fieldPath", fieldPath, "message", msg)
+			return typez
+		}
+		for _, field := range message.Fields {
+			if name != field.Name {
+				continue
+			}
+			typez = field.Typez
+			if field.Typez == api.MESSAGE_TYPE {
+				msg = state.MessageByID[field.TypezID]
```

Comment:
I am confused by what we are doing with `msg` here. Is it supposed to be recursive?

Reply:
Yes, it was.  Fixed.

My tests were completely ineffective because they reused the field names. Fixed those too.

Reply:
>  they reused the field names. 

Ah, I could not explain why some of the test cases worked.